### PR TITLE
dark magic v0.5.0: persistent user state

### PR DESCRIPTION
**Pull request by jhheider**
@mxcl this is a fun one. fundamental direction being implemented here. thoughts more than welcome.

### CODE REVIEW

**Comment by mxcl** [comment]
@jhheider I dare not resolve this one

**Comment by mxcl** [comment]
who provides sqlite? deno has it embedded?

File: src/hooks/useDb.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+export default class Db {
+  db: SSQL
+  static models = [
+    Config,
+    DarkMagicChoice,
+  ]
+
+  constructor(db: Path | undefined = undefined) {
+    const db_ = db ?? Path.home().join(".local/state/tea/db.sqlite3")
```
[the remark this answers is not in the document]
Comment (jhheider) [reply src/hooks/useDb.ts:25]: Sure can. Worth adding to Path.ts?

[the remark this answers is not in the document]
Comment (mxcl) [reply src/hooks/useDb.ts:25]: that seems like a good addition yes

**Comment by mxcl** [comment]
who provides sqlite? deno has it embedded?

**Comment by jhheider** [comment]
> who provides sqlite? deno has it embedded?

Good question. Could be pure js/ts.

**Comment by jhheider** [comment]
```sh
$ otool -l ./tea | grep sqli
$
```

https://github.com/enimatek-nl/small-orm-sqlite/blob/2d6ef6a10b32c0b7505232ff909d75468a67b9c2/lib/SmallSQLite.ts#L1

Ah, right; I remember reading [this](https://github.com/dyedgreen/deno-sqlite#deno-sqlite-module):  `This is an SQLite module for JavaScript and TypeScript. The wrapper is targeted at [Deno](https://deno.land/) and uses a version of SQLite3 compiled to WebAssembly (WASM). This module focuses on correctness, ease of use and performance.`

File: src/hooks/useDb.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhheider) [reply src/hooks/useDb.ts]: I like how that came out. feels like encouraging best practices.

**Comment by mxcl** [comment]
Ah I wonder how it impacts our binary size. Well we can sort that out some other time.

**Comment by jhheider** [comment]
```sh
jacob@illithid cli % ls -lh ./tea ~/.tea/tea.xyz/v0.26.0/bin/tea 
-rwxrwxrwx@ 1 jacob  staff    77M Apr 11 15:22 ./tea
-rwxr-xr-x  1 jacob  staff    80M Mar 27 15:17 /Users/jacob/.tea/tea.xyz/v0.26.0/bin/tea
```
possibly minimally.

**Comment by jhheider** [comment]
exciting. now we have a state store available!

**Pull request merged**